src/pages/Dashboard.tsx

```diff
@@ -176,6 +154,50 @@ const Dashboard = () => {
         };
     };
 
+    const [isTaskModelOpen, setIsTaskModelOpen] = useState<boolean>(false);
+    const handleOpenTaskModal = () => setIsTaskModelOpen(true);
+    const handleCloseTaskModal = () => setIsTaskModelOpen(false);
+
+    const [currentTaskModalMode, setCurrentTaskModalMode] = useState<ModalMode>('Create');
+
+    const [currentEditTaskId, setCurrentEditTaskId] = useState<string>('');
+    const [currentEditTask, setCurrentEditTask] = useState<Task | undefined>();
+
+    const [loadingTaskModal, setLoadingTaskModal] = useState<boolean>(false);
+
+    const generateEditTaskHandler = (taskId: string, task: Task): (() => void) => {
+        return () => {
+            setCurrentEditTaskId(taskId);
+            setCurrentEditTask({ ...task });
+            setCurrentTaskModalMode('Edit');
+            setLoadingTaskModal(true);
+        };
+    };
+
+    const openNewTaskModal = () => {
+        setCurrentTaskModalMode('Create');
+        setLoadingTaskModal(true);
+    };
+
+    useEffect(() => {
+        if (loadingTaskModal) {
+            handleOpenTaskModal();
+            setLoadingTaskModal(false);
+        }
+    }, [loadingTaskModal]);
+
+    const editTask = async (task: Task) => {
+        try {
+            await updateTask(TEST_ORGANIZATION, currentViewedGroup, currentEditTaskId, task);
+            const tasksCopy: { [id: string]: Task } = { ...displayedTasks };
```

Comment:
Just a trick but you can setDisplayedTasks like
```
setDisplayedTasks((prevDisplayedTasks) => prevDisplayedTasks.map((task, index) => {
    index = currentEditTaskId ? return newTask: return
}
```

or something, please check my code for accuracy.